opensquirrel/squirrel_ir.py

```diff
@@ -221,6 +225,20 @@ def is_identity(self) -> bool:
         return self.target_gate.is_identity()
 
 
+def _compare_gate_classes(g1: ControlledGate | MatrixGate, g2: ControlledGate | MatrixGate) -> bool:
+
+    from opensquirrel.replacer import get_replacement_matrix
+
+    union_mapping = list(set(g1.get_qubit_operands()) | set(g2.get_qubit_operands()))
+
+    matrix_g1 = get_replacement_matrix([g1], union_mapping)
+    matrix_g2 = get_replacement_matrix([g2], union_mapping)
+    if matrix_g1.shape != matrix_g2.shape:
+        return False
```

Comment:
this `return False` will never be executed, because the mappings are identical
the matrix returned by `get_replacement_matrix` has shape completely determined by the `len` of the mapping